woocommerce/woocommerce
# [CYS] Fix AI generated site title comparison

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





I noticed the AI-generated site title was not updating properly in my site and the reason was html encoding:
It was comparing `Book &amp; Rose Emporium` with `Book & Rose Emporium`, which was failing thus never updating the site title again. This PR addresses this issue using the `html_entity_decode`.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you are on a WooCommerce site where the CYS AI flow works.
2. Go to `WooCommerce > Home > Customize your store`.
3. Click on `Design With AI` and proceed to the assembler.
4. Check the title is correctly generated and related to the description you used.
5. Repeat the process with a different store prompt and make sure the title is regenerated.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
[CYS] Fix the AI-generated site title comparison.
#### Comment 

</details>
